tests/vmi_multus_test.go

```diff
@@ -58,6 +58,8 @@ const (
 	sriovConfCRD           = `{"apiVersion":"k8s.cni.cncf.io/v1","kind":"NetworkAttachmentDefinition","metadata":{"name":"%s","namespace":"%s","annotations":{"k8s.v1.cni.cncf.io/resourceName":"%s"}},"spec":{"config":"{ \"cniVersion\": \"0.3.1\", \"name\": \"sriov\", \"type\": \"sriov\", \"ipam\": { \"type\": \"host-local\", \"subnet\": \"10.1.1.0/24\" } }"}}`
 	sriovLinkEnableConfCRD = `{"apiVersion":"k8s.cni.cncf.io/v1","kind":"NetworkAttachmentDefinition","metadata":{"name":"%s","namespace":"%s","annotations":{"k8s.v1.cni.cncf.io/resourceName":"%s"}},"spec":{"config":"{ \"cniVersion\": \"0.3.1\", \"name\": \"sriov\", \"type\": \"sriov\", \"link_state\": \"enable\", \"ipam\": { \"type\": \"host-local\", \"subnet\": \"10.1.1.0/24\" } }"}}`
 	macvtapNetworkConf     = `{"apiVersion":"k8s.cni.cncf.io/v1","kind":"NetworkAttachmentDefinition","metadata":{"name":"%s","namespace":"%s", "annotations": {"k8s.v1.cni.cncf.io/resourceName": "macvtap.network.kubevirt.io/%s"}},"spec":{"config":"{ \"cniVersion\": \"0.3.1\", \"name\": \"%s\", \"type\": \"macvtap\"}"}}`
+	sriovConfVlanCRD       = `{"apiVersion":"k8s.cni.cncf.io/v1","kind":"NetworkAttachmentDefinition","metadata":{"name":"%s","namespace":"%s","annotations":{"k8s.v1.cni.cncf.io/resourceName":"%s"}},"spec":{"config":"{ \"cniVersion\": \"0.3.1\", \"name\": \"sriov\", \"type\": \"sriov\", \"link_state\": \"enable\", \"vlan\": 200, \"ipam\": {}}"}}`
+	sriovConfNoVlanCRD     = `{"apiVersion":"k8s.cni.cncf.io/v1","kind":"NetworkAttachmentDefinition","metadata":{"name":"%s","namespace":"%s","annotations":{"k8s.v1.cni.cncf.io/resourceName":"%s"}},"spec":{"config":"{ \"cniVersion\": \"0.3.1\", \"name\": \"sriov\", \"type\": \"sriov\", \"link_state\": \"enable\", \"vlan\": 0, \"ipam\": {}}"}}`
```

Comment:
> Why IPAM is not used?

Maybe @yossisegev  can answer this. I just followed the test plan. 

> Why do we need to specify vlan: 0 explicitly? Either do not specify anything or give a different vlan id.
> I mainly want to understand why we need to support 2 new NAD/s instead of a single additional one.

When using the sriov-operator, and you create `sriovNetwork` object with no vlan, [it gives a default value of 0](https://docs.openshift.com/container-platform/4.2/networking/multiple_networks/configuring-sr-iov.html#configuring-sr-iov-networks_configuring-sr-iov) to the vlan field in the NAD. In order for the test to mock the common use case I thought this is necessary.

So, since no IPAM, and vlan=0, I figured it's best to create another two NADs .

Reply:
What test plan?

IPAM is not relevant in all SRIOV NAD/s, so we need to clean the old ones but for this context, you do not need a new NAD for sure.

> So, since no IPAM, and vlan=0, I figured it's best to create another two NADs .

As IPAM is a noop here and we can re-use the existing NAD, lets use the existing.

Reply:
Here is the test plan:  https://polarion.engineering.redhat.com/polarion/#/project/CNV/workitem?id=CNV-4713

Thank you for the info, I also confirmed it with @yossisegev  that there is no reason to not use IPAM. so I added it to the new NAD.

So now, non-vlan-vmi is connected to the existing sirov network, and only one NAD is created.

If you think there is a need to use vlan=0, in order to mock the sriov-operator, then I can change it back, but I will have to use a different NAD.